microstream/src/main/java/io/micronaut/microstream/health/MicrostreamHealthIndicator.java

```diff
@@ -0,0 +1,117 @@
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A {@link HealthIndicator} that checks the health of all registered {@link EmbeddedStorageManager}s.
+ *
+ * @since 1.0.0
+ * @author Tim Yates
+ */
+@Singleton
+@Requires(classes = HealthIndicator.class)
+@Requires(property = HealthEndpoint.PREFIX + ".microstream.enabled", value = StringUtils.TRUE, defaultValue = StringUtils.TRUE)
+public class MicrostreamHealthIndicator implements HealthIndicator {
+
+    public static final String MICROSTREAM_PREFIX = "microstream.";
+    private final Map<String, EmbeddedStorageManager> embeddedStorageManagerMap = new ConcurrentHashMap<>();
+
+    public MicrostreamHealthIndicator(BeanContext beanContext) {
+        for (BeanDefinition<EmbeddedStorageManager> definition : beanContext.getBeanDefinitions(EmbeddedStorageManager.class)) {
+            if (definition.getDeclaredQualifier() instanceof Named) {
+                String name = ((Named) definition.getDeclaredQualifier()).getName();
+                EmbeddedStorageManager embeddedStorageManager = beanContext.getBean(definition);
+                embeddedStorageManagerMap.putIfAbsent(name, embeddedStorageManager);
+            }
+        }
+    }
+
+    @Override
+    public Publisher<HealthResult> getResult() {
+        return Flux.fromIterable(embeddedStorageManagerMap.entrySet())
+            .map(namedBean ->
+                HealthResult.builder(
+                        MICROSTREAM_PREFIX + namedBean.getKey(),
+                        namedBean.getValue().isRunning() ? HealthStatus.UP : HealthStatus.DOWN
+                    )
+                    .details(new HealthHolder(namedBean.getValue()))
+                    .build()
+            );
+    }
+
+    @Introspected
+    @SuppressWarnings("unused")
+    static class HealthHolder {
+
+        private final boolean startingUp;
+        private final boolean running;
+        private final boolean active;
+        private final boolean acceptingTasks;
+        private final boolean shuttingDown;
+        private final boolean shutdown;
+
+        HealthHolder(EmbeddedStorageManager manager) {
+            this.startingUp = manager.isStartingUp();
+            this.running = manager.isRunning();
+            this.active = manager.isActive();
+            this.acceptingTasks = manager.isAcceptingTasks();
+            this.shuttingDown = manager.isShuttingDown();
+            this.shutdown = manager.isShutdown();
+        }
+
+        public boolean isStartingUp() {
+            return startingUp;
+        }
+
+        public boolean isRunning() {
+            return running;
+        }
+
+        public boolean isActive() {
+            return active;
+        }
+
+        public boolean isAcceptingTasks() {
+            return acceptingTasks;
+        }
+
+        public boolean isShuttingDown() {
+            return shuttingDown;
+        }
+
+        public boolean isShutdown() {
+            return shutdown;
+        }
+    }
+}
```

Comment:
I wanted this to have `public final` fields, and no getters, but then `checkstyle` fails

Any idea why we have this checkstyle rule?

Am I ok to ignore the rule for this class?  It's just for serialization

Reply:
Changing it to this is functionally identical

```java
    @Introspected
    @SuppressWarnings({"unused", "checkstyle:VisibilityModifier"})
    private static class HealthHolder {

        public final boolean startingUp;
        public final boolean running;
        public final boolean active;
        public final boolean acceptingTasks;
        public final boolean shuttingDown;
        public final boolean shutdown;

        HealthHolder(EmbeddedStorageManager manager) {
            this.startingUp = manager.isStartingUp();
            this.running = manager.isRunning();
            this.active = manager.isActive();
            this.acceptingTasks = manager.isAcceptingTasks();
            this.shuttingDown = manager.isShuttingDown();
            this.shutdown = manager.isShutdown();
        }
    }
```

Reply:
I have switched to use the shorter (more readable IMHO) version

Reply:
Resolved via: https://github.com/micronaut-projects/micronaut-microstream/pull/22/commits/bf6d8fdc124aa7322a0e32540f88f93d86560018